python/dask_cudf/dask_cudf/io/csv.py

```diff
@@ -1,8 +1,193 @@
 # Copyright (c) 2024, NVIDIA CORPORATION.
 
-from dask_cudf import _deprecated_api
+import os
+from glob import glob
+from warnings import warn
 
-read_csv = _deprecated_api(
-    "dask_cudf.io.csv.read_csv",
-    new_api="dask_cudf.read_csv",
-)
+from fsspec.utils import infer_compression
+
+from dask import dataframe as dd
+from dask.dataframe.io.csv import make_reader
+from dask.utils import parse_bytes
+
+import cudf
+
+
+def read_csv(path, blocksize="default", **kwargs):
+    """
+    Read CSV files into a :class:`.DataFrame`.
+
+    This API parallelizes the :func:`cudf:cudf.read_csv` function in
+    the following ways:
+
+    It supports loading many files at once using globstrings:
+
+    >>> import dask_cudf
+    >>> df = dask_cudf.read_csv("myfiles.*.csv")
+
+    In some cases it can break up large files:
```

Comment:
What cases are those? Or is it always dependent upon the file size and the value of `blocksize`? If the latter, maybe we should just rephrase to "It can also break up large files by specifying the size of each block via `blocksize`".

Reply:
Sorry, I didn't spend any time reviewing these doc-strings, because they were directly copied from `dask_cudf/_legacy/io/csv.py`. Does it make sense to address these suggestions/questions in a follow-up (just to make sure CI is unblocked)?